### PR TITLE
change scroll lock module

### PR DESCRIPTION
**Pull request by yuki0410-dev**
change scroll lock module from [disable-scroll](https://www.npmjs.com/package/disable-scroll) to [body-scroll-lock](https://www.npmjs.com/package/body-scroll-lock).

## Change
- remove  disable-scroll
- add body-scroll-lock
- Moved Modal Option processing
- add clean up when unmouted. (in useEffect)

## Why
- disable-scroll trap keydown and  can't customize target element ([source](https://github.com/gilbarbara/disable-scroll/blob/9608b91d03ccb88298aecf41551aa9054968d9e4/src/index.ts#L93))
  - #27 
- body-scroll-lock not  trap keydown.
  - it is more popular module ([npm trends](https://www.npmtrends.com/body-scroll-lock-vs-disable-scroll-vs-dom-locky-vs-scroll-lock))

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -50,39 +51,64 @@ const containerStyle: React.CSSProperties = {
...
 };
 
-const Modal: React.FC<ModalProps> = ({ children, isOpen = false, close, elementId = 'root' }) => {
+const Modal: React.FC<ModalProps> = ({ children, isOpen = false, close, elementId = 'root', options = {} }) => {
+  const { preventScroll = false } = options;
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const onOpen = useCallback(() => {
+    if (preventScroll) {
+      if (modalRef.current != null) {
```
Comment (shibe97) [inline src/index.tsx:60]: `!==` is better.

File: src/index.tsx [hunk 2]
```diff
@@ -50,39 +51,64 @@ const containerStyle: React.CSSProperties = {
...
+        disableBodyScroll(modalRef.current, {
+          reserveScrollBarGap: true,
+        });
+      }
+    }
+  },[preventScroll, modalRef.current]);
+
+  const onClose = useCallback(() => {
+    if (preventScroll) {
+      if (modalRef.current != null) {
```
Comment (shibe97) [inline src/index.tsx:70]: `!==` is better.

File: src/index.tsx (hunk 1 above)
Comment (ohnaka0410) [reply src/index.tsx]: @shibe97 
Thank you for reviewing.
I used it to mean `!== null` and `!== undefined`.
The condition of `undefined` cannot occur, so it has been fixed.

**Comment by dc7290** [comment]
@yuki0410-dev 

Thank you！
I also apologize that it has been quite some time.

LGTM!

**Pull request merged**